File: Frontend/angular-ecommerce/src/app/common/purchase.ts
import { Address } from "./address";
import { Customer } from "./customer";
import { OrderItem } from "./order-item";
import { Orders } from "./orders";

export class Purchase {
            customer: Customer=new Customer();
            shippingAddress: Address=new Address();
            billingAddress: Address=new Address();
            orders: Orders=new Orders();
            orderItems: OrderItem[]=[];
}

File: Backend/spring-boot-ecommerce/spring-boot-ecommerce/src/main/java/com/prash/ecommerce/dto/Purchase.java
package com.prash.ecommerce.dto;

import com.prash.ecommerce.entity.Address;
import com.prash.ecommerce.entity.Customer;
import com.prash.ecommerce.entity.OrderItem;
import com.prash.ecommerce.entity.Order;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order orders;
    private Set<OrderItem> orderItems;
}
